app/src/main/java/com/amaze/filemanager/activities/ImageViewerActivity.java

```diff
@@ -0,0 +1,315 @@
+    }
+  }
+
+  private void loadFileList() {
+    int IS_FOLDER = 1;
+    if (FileUtil.checkFolder(currentFile.getParent(this), this) == IS_FOLDER) {
+      BaseFile parentDir = new BaseFile(currentFile.getParent(this));
+      baseFiles = filterList(parentDir.listFiles(this, parentDir.isRoot()));
+    }
+  }
+
+  private ArrayList<BaseFile> filterList(ArrayList<BaseFile> baseFiles) {
+    ArrayList<BaseFile> newList = new ArrayList<>();
+
+    for (int i = 0; i < baseFiles.size(); i++) {
+      BaseFile baseFile = baseFiles.get(i);
+      Log.d(TAG, "filterList: " + getMimeType(baseFile.getPath()));
+      String mimeType = "";
+      if (getMimeType(baseFile.getPath()) != null) {
+        mimeType = getMimeType(baseFile.getPath());
+      }
+      if (mimeType.contains("image/")) {
+        newList.add(baseFile);
+      }
+
+      if (baseFile.getPath().equals(currentFile.getPath())) {
+        currentIndex = i;
+      }
+    }
+    return newList;
+  }
+
+  private void initToolbar() {
+    toolbar = (Toolbar) findViewById(R.id.toolbar);
+    setSupportActionBar(toolbar);
+  }
+
+  private void initIvImageView() {
+    ivImage = (TouchImageView) findViewById(R.id.ivImage);
+    loadImage();
+
+    ivImage.setOnTouchListener(new View.OnTouchListener() {
+      @Override public boolean onTouch(View v, MotionEvent event) {
+        if (ivImage.getCurrentZoom() > NO_ZOOM) {
+          return false;
+        }
+
+        switch (event.getAction()) {
+          case MotionEvent.ACTION_DOWN:
+
+            oldXValue = event.getX();
+            break;
+
+          case MotionEvent.ACTION_UP:
+            if (ivImage.getCurrentZoom() != NO_ZOOM) {
+              return false;
+            }
+            float currentX = event.getX();
+
+            if (currentX > oldXValue) {
+              //SWIPE RIGHT
+              showPreviousImage();
+            }
+
+            if (currentX < oldXValue) {
+              //SWIPE LEFT
+              showNextImage();
+            }
+
+            break;
+        }
+        return true;
+      }
+    });
+
+    ivImage.setOnClickListener(new View.OnClickListener() {
+      @Override public void onClick(View v) {
+
+        if (llToolbarLayout.getVisibility() == View.VISIBLE) {
+          setMenuVisibility(View.INVISIBLE);
+        } else {
+          setMenuVisibility(View.VISIBLE);
+        }
+      }
+    });
+  }
+
+  private void setMenuVisibility(int visible) {
+    llToolbarLayout.setVisibility(visible);
+    llBottomBar.setVisibility(visible);
+  }
+
+  private void loadImage() {
+    ivImage.loadImageWithGlide(currentFile.getPath());
+  }
+
+  private void showNextImage() {
+    if (currentIndex < baseFiles.size() - 1) {
+      currentIndex++;
+      currentFile = baseFiles.get(currentIndex);
+      loadImage();
+      updateTvFolderName();
+    }
+  }
+
+  private void updateTvFolderName() {
+    tvFolderName.setText("Folder:"
```

Comment:
Hardcoded "Folder:", are you sure? Also check [this](https://stackoverflow.com/questions/3656371/dynamic-string-using-string-xml) out.